### PR TITLE
Catch ChannelNotFoundEntity and raise as RasaException

### LINKED ISSUE #8178: use `RasaException` for `ChannelNotFoundEntity` exception
Sentry Issue: [RASA-OPEN-SOURCE-78E](https://sentry.io/organizations/rasahq/issues/2260937721/?referrer=github_integration)

```
CancelledError: 
  File "aiormq/base.py", line 25, in __inner
    return await self.task

ChannelNotFoundEntity: NOT_FOUND - home node 'rabbit@rabbit-rabbitmq-0.rabbit-rabbitmq-headless.rabbitmq.svc.cluster.local' of durable queue 'rasa_production_events' in vhost '/' is down or inaccessible
(21 additional frame(s) were not displayed)
...
  File "rasa/core/brokers/broker.py", line 84, in _create_from_endpoint_config
    broker = await PikaEventBroker.from_endpoint_config(endpoint_config, event_loop)
  File "rasa/core/brokers/pika.py", line 142, in from_endpoint_config
    await broker.connect()
  File "rasa/core/brokers/pika.py", line 157, in connect
    self._exchange = await self._set_up_exchange(channel)
  File "rasa/core/brokers/pika.py", line 213, in _set_up_exchange
    await asyncio.gather(
  File "rasa/core/brokers/pika.py", line 226, in _bind_queue
    queue = await channel.declare_queue(queue_name, durable=True)
```

**Definition of Done**
- [ ] This exception is caught and re-raised as `RasaExeption` since it's a usage error

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Fix #8178 

**Status (please check what you already did)**:
- [X] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/core/test_broker.py [hunk 1]
```diff
@@ -96,6 +99,16 @@ def test_pika_queues_from_args(
...
 
 
+async def test_pika_raise_rasa_exception():
+    pika_broker = PikaEventBroker("host", "username", "password")
+    mock_channel = AsyncMock()
+    mock_channel.declare_queue = AsyncMock(side_effect=ChannelNotFoundEntity())
+    mock_exchange = AsyncMock()
+
+    with pytest.raises(RasaException):
+        await pika_broker._bind_queue("test_queue", mock_channel, mock_exchange)
```
Comment (joejuzl) [inline tests/core/test_broker.py:109]: Is it possible to test this by calling a public method? It's always preferable to test from the public API of an object.

**Review by joejuzl — CHANGES_REQUESTED** [review]

File: rasa/core/brokers/pika.py [hunk 2]
```diff
@@ -223,7 +225,11 @@ async def _set_up_exchange(
     async def _bind_queue(
         queue_name: Text, channel: aio_pika.RobustChannel, exchange: aio_pika.Exchange
     ) -> None:
-        queue = await channel.declare_queue(queue_name, durable=True)
+        try:
+            queue = await channel.declare_queue(queue_name, durable=True)
+        except ChannelNotFoundEntity as e:
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:230]: How about handling the exception [here](https://github.com/RasaHQ/rasa/blob/5a9b83bacda02f0a293ef40c7e394afdc82de868/rasa/core/brokers/broker.py#L33) together with the others? Would having to duplicate the logic and makes sure we're handling exceptions upon the broker creation consistently. I'd also throw `ConnectionException` (it's a subclass of `RasaException`) instead of `RasaException` directly as it's more specific.

File: tests/core/test_broker.py (hunk 1 above)
Comment (ancalita) [reply tests/core/test_broker.py]: I've edited the test after applying Tobias's suggestion to add the `ChannelNotFoundEntity` exception to `EventBroker `create method. Please check if everything looks ok.

File: rasa/core/brokers/pika.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 from typing import Deque, Dict, Optional, Text, Union, Any, List, Tuple
 
 import aio_pika
+from aiormq.exceptions import ChannelNotFoundEntity
```
Comment (joejuzl) [inline rasa/core/brokers/pika.py:11]: I think we can revert the changes in this file now right?

**Review by joejuzl — CHANGES_REQUESTED** [review]

File: rasa/core/brokers/pika.py (hunk 2 above)
Comment (joejuzl) [reply rasa/core/brokers/pika.py]: I think for consistency it makes sense to have it where you suggest. But in general it is better not to leak specific implementation exceptions into the base class.

File: rasa/core/brokers/pika.py (hunk 3 above)
Comment (ancalita) [reply rasa/core/brokers/pika.py]: Done 😄

Comment (joejuzl) [reply rasa/core/brokers/pika.py]: there's still a couple more in the file 😄

Comment (ancalita) [reply rasa/core/brokers/pika.py]: oups, found it - all removed now I hope 😅

File: changelog/8178.bugfix.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Caught ChannelNotFoundEntity exception and re-raised as RasaException.
```
Comment (joejuzl) [inline changelog/8178.bugfix.md:1]: This needs to be updated. Also might be worth mentioning the error is coming from the pika broker.

File: tests/core/test_broker.py [hunk 5]
```diff
@@ -96,6 +99,18 @@ def test_pika_queues_from_args(
     assert pika_processor.queues == expected
 
 
+async def test_pika_raise_rasa_exception(monkeypatch: MonkeyPatch):
+
+    monkeypatch.setattr(
+        PikaEventBroker, "connect", AsyncMock(side_effect=ChannelNotFoundEntity())
+    )
+
+    with pytest.raises(RasaException):
```
Comment (joejuzl) [inline tests/core/test_broker.py:108]: This works because `RasaException` is a subclass of `ConnectionException`, but maybe better to test that `ConnectionException` is raised?

Comment (ancalita) [reply tests/core/test_broker.py:108]: Actually, `ConnectionException` is a subclass of `RasaException`, I've tested locally with `ConnectionException` - all seems fine so I'll make this edit.

Comment (joejuzl) [reply tests/core/test_broker.py]: > Actually, ConnectionException is a subclass of RasaException
Yup sorry that's what I meant - wouldn't work if it was the other way around.

**Review by joejuzl — APPROVED** [review]
Looks great! 🚀

File: changelog/8178.bugfix.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Catch ChannelNotFoundEntity exception coming from the pika broker and raise as ConnectionException.
```
Comment (m-vdb) [inline changelog/8178.bugfix.md:1]: @ancalita also worth using some markdown formatting for these in the future, for instance using backticks:
```
Catch `ChannelNotFoundEntity` exception coming from the pika broker and raise as `ConnectionException`.
```

That's probably nitpicking at this stage (sorry I was just passing by on your PR 😅 ), and I am not sure we even have guidelines for changelog fragments formatting 🤔 . I know we have [this page](https://www.notion.so/rasa/Docs-writing-guidelines-8c0ed021fb524f348bf94f9426439c86) documenting how we should write documentation pages though (worth taking a look at some point).
